extensions/amp-list/0.1/amp-list.js

```diff
@@ -809,6 +832,75 @@ export class AmpList extends AMP.BaseElement {
     });
   }
 
+  /**
+   * Updates `this.container_` by DOM diffing its children against `elements`.
+   * @param {!Element} container
+   * @param {!Array<!Element>} elements
+   * @private
+   */
+  diff_(container, elements) {
+    const newContainer = this.createContainer_();
+    this.addElementsToContainer_(elements, newContainer);
+
+    // Typically, diff-marking elements happens during template sanitization.
+    // This obviously doesn't apply to initial content, so we mark them manually
+    // here to enable diffing in the first render.
+    if (this.hasInitialContent_) {
+      // amp-mustache starts at 1 and increments, so start at -1 and decrement
+      // to guarantee uniqueness.
+      let key = -1;
+      // We only need to mark AMP elements for diffing because bindings in
+      // initial content are inert.
+      const elements = container.querySelectorAll('.i-amphtml-element');
+      elements.forEach(element => {
+        markElementForDiffing(element, () => String(key--));
+      });
+    }
+
+    const ignored = setDOM(container, newContainer);
+
```

Comment:
Nit: Could we break this large `diff` function down a bit so that the code can be readable without so many comments? =)

Reply:
Done. :)